src/appleseed/foundation/math/sampling/qmcsamplingcontext.h

```diff
@@ -381,6 +390,42 @@ inline Vector<T, N> QMCSamplingContext<RNG>::next2(Tag<Vector<T, N>>)
     return v;
 }
 
+template <typename RNG>
+template <typename T>
+inline std::vector<Vector<T, 2>> QMCSamplingContext<RNG>::next2(const size_t N)
+{
+	std::vector<Vector<T, 2>> v(N);
+
+    assert(m_sample_count == 0 || m_instance < m_sample_count);
+    assert(m_dimension == 2);
+    assert(N <= PrimeTableSize);
+
+    if (m_mode == QMCMode)
+    {
+        // Correlated Multi-Jittered Sampling
+        unsigned int m, n;
+        cmj_compute_mn(m, n, N);
+
+        for (size_t i = 0; i < N; ++i)
+        {
+            v[i] = cmj_generate_sample<T>(i, m, n, N, m_instance);
+        }
+    }
+    else
+    {
+        for (size_t i = 0; i < N; ++i)
+        {
+            v[i] = Vector<T, 2>(
```

Comment:
There's a subtle error here: C++ doesn't mandate any specific order for arguments evaluation, hence the final vector may have different values on different compilers, compiler versions, optimization levels, etc. To make the code fully deterministic (which is a requirement in appleseed) you must do

    v[i].x = rand2<T>(m_rng);
    v[i].y = rand2<T>(m_rng);

Reply:
Weird behavior :o